File: rust/axum/src/modules/manager/repository.rs
use super::entity::ManagerEntity;
use crate::database;
use sqlx::Error;
pub struct ManagerRepository;

impl ManagerRepository {
	pub async fn find_by_email(email: &str) -> Result<Option<ManagerEntity>, Error> {
		let manager = sqlx::query_as::<_, ManagerEntity>(
			r#"
			SELECT * FROM "Gerente" WHERE email = $1 LIMIT 1
			"#
		)
		.bind(email)
		.fetch_optional(database::db())
		.await;

		match manager {
			Ok(manager) => Ok(manager),
			Err(err) => Err(err),
		}
	}

	pub async fn find_all(quantidade: i32) -> Result<Vec<ManagerEntity>, Error> {
		let managers = sqlx::query_as::<_, ManagerEntity>(
			r#"
			SELECT * FROM "Gerente" LIMIT $1
			"#
		)
    .bind(quantidade)
    .fetch_all(database::db())
    .await;

    match managers {
      Ok(managers) => Ok(managers),
      Err(err) => Err(err),
    }
	}
}

File: go/fiber_orm/internal/modules/manager/repository.go
package manager

import (
	"tcc/internal/database"
	"tcc/internal/modules/manager/entity"
)

type ManagerRepository struct {
}

func (r *ManagerRepository) FindByEmail(email string) (*entity.ManagerEntity, error) {

	var manager entity.ManagerEntity
	result := database.Conn.Where("email = ?", email).First(&manager)

	if result.Error != nil {
		return nil, result.Error
	}

	return &manager, nil
}


func (r *ManagerRepository) FindAll(quantidade int) ([]*entity.ManagerEntity, error) {

	var managers []*entity.ManagerEntity
	result := database.Conn.Limit(quantidade).Find(&managers)

	if result.Error != nil {
		return nil, result.Error
	}

	return managers, nil
}